ifrs17/Report/ParameterReportMutableScopeInteractive.ipynb

```diff
@@ -0,0 +1,246 @@
+{
+    "metadata": {
+        "authors": [],
+        "id": "XdWtZAfDUkKgJxSlrhCSSw",
+        "kernelspec": {
+            "display_name": "Formula Framework",
+            "language": "C#",
+            "name": "C#"
+        },
+        "language_info": {
+            "file_extension": ".cs",
+            "mimetype": "text/plain",
+            "name": "C#"
+        }
+    },
+    "nbformat": 4,
+    "nbformat_minor": 5,
+    "cells": [
+        {
+            "cell_type": "code",
+            "source": [
+                "#!import \"ReportMutableScopesInteractive\"",
+                "\n#!import \"ParameterReportsQueries\""
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "public interface RepFormsEntity : MutableScopeWithWorkspace",
+                "\n{",
+                "\n    [DropdownMethod(nameof(ReportMenu))]",
+                "\n    string ReportType{get; set;}",
+                "\n",
+                "\n    IReadOnlyCollection<string> ReportMenu() => new string[] {\"\", \"DataNodeReport\", ",
+                "\n                                                            \"DataNodeStateReport\", ",
+                "\n                                                            \"YieldCurves\", ",
+                "\n                                                            \"SingleDataNodeReport\", ",
+                "\n                                                            \"InterDataNodeParameters\", ",
+                "\n                                                            \"PartnerRating\", ",
+                "\n                                                            \"PartnerDefaultRates\"};",
+                "\n}"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "public interface ParameterReportScope: IMutableScope<string>, ReportingNodeFormsEntity, MonthlyPeriodFormsEntity, ScenarioFormsEntity, CurrencyFormsEntity, RepFormsEntity",
+                "\n{",
+                "\n    protected IPivotFactory report => GetStorage().Report;",
+                "\n    protected IExportVariable export => GetStorage().Export;",
+                "\n    protected int headerColumnWidthValue => 250;",
+                "\n",
+                "\n    HashSet<(ReportIdentity, CurrencyType)> GetDataIdentities() => GetStorage().GetIdentities((Year, Month), ReportingNode, Scenario, CurrencyType); ",
+                "\n",
+                "\n    async Task InitializeStorageAsync()",
+                "\n    {",
+                "\n        await GetStorage().InitializeReportIndependentCacheAsync();",
```

Comment:
Shouldn't this be in the INIT method of the scope, as you should do it only once (this should be independent from args).

Reply:
OK, I found the workaround to initialize the scope with non-awaited routine  along the lines of:

var task = ....Async();
task.Wait();

This workaround works, so I left it here. Please take a look.